providers/shared/flows/context/context.ftl

```diff
@@ -751,14 +753,131 @@ The roles available are defined as part of the state of the target component.
+        [/#if]
+        [#return
+            internalCreateOccurrenceFromContextExternalLink(occurrence, link) +
+            {
+                "Direction" : fullLink.Direction?lower_case,
+                "Role" : fullLink.Role,
+                "IncludeInContext" : fullLink.IncludeInContext![]
+            }
+        ]
+    [/#if]
+
+    [#-- NOTE: Filter must contain all relevant attributes In particular, links            --]
+    [#-- to lambda functions much contain the function attribute even if only one function --]
+    [#local targetContextLists = findContexts(model, getLinkAsMatch(fullLink)) ]
+
+    [#if targetContextLists?size > 1]
+        [@fatal
+            message="Link matched multiple targets"
+            context=targetContextLists
+        /]
+        [#return {} ]
+    [/#if]
+
+    [#if targetContextLists?has_content]
+        [@debug
+            message="Link matched target"
+            context=targetContextLists[0]
+            enabled=false
+        /]
+
+        [#local targetOccurrence = internalCreateOccurrenceFromContextLists([targetContextLists[0]]) ]
+        [@debug
+            message="TargetOccurrence"
+            context=targetOccurrence
+            enabled=false
+        /]
+
+        [#-- Determine if deployed --]
+        [#if ( activeOnly || activeRequired ) && !isOccurrenceDeployed(targetOccurrence) ]
+            [#if activeRequired ]
+                [@postcondition
+                    function="getContextLinkTarget"
+                    context=
+                        {
+                            "Occurrence" : occurrence,
+                            "Link" : link,
+                            "FullLink" : fullLink
+                        }
+                    detail="COTFatal:Link target not active/deployed"
+                /]
+            [/#if]
+            [#return {} ]
+        [/#if]
+
+        [#-- Determine the role --]
+        [#local role =
+            fullLink.Role!getOccurrenceDefaultRole(targetOccurrence, fullLink.Direction?lower_case)]
+
+        [#return
+            targetOccurrence +
+            {
+                "Direction" : fullLink.Direction?lower_case,
+                "Role" : role,
+                "IncludeInContext" : fullLink.IncludeInContext![]
+            } ]
+    [/#if]
+
+    [@postcondition
+        function="getContextLinkTarget"
+        context=
+            {
+                "Occurrence" : occurrence,
+                "Link" : link,
+                "FullLink" : fullLink
+            }
+        detail="COTFatal:Link not found"
+    /]
+    [#return {} ]
+[/#function]
+
 [#function createLink id name filter={} direction="outbound" role="" attributes={} enabled=true]
     [#return
         {
             "Id" : id,
             "Name" : name,
             "Enabled" : enabled,
             "Filter" : filter,
-            "Direction" : direction?lower_case,
+            "Direction" : direction,
```

Comment:
need to pick up this change from previously merged PR

Reply:
Fixed, thought I had caught all of that